assets/src/admin/paired-browsing/app.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * Internal dependencies
+ */
+import './app.css';
+
+const { app, history } = window;
+const { ampSlug, ampPairedBrowsingQueryVar, ampValidationErrorsQueryVar } = app;
+
+class PairedBrowsingApp {
+	exitLink;
+
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		this.nonAmpIframe = document.querySelector( '#non-amp iframe' );
+		this.ampIframe = document.querySelector( '#amp iframe' );
+		this.ampPageHasErrors = false;
+
+		// Link to exit paired browsing.
+		this.exitLink = document.getElementById( 'exit-link' );
+
+		// Overlay that is displayed on the client that becomes disconnected.
+		this.disconnectOverlay = document.querySelector( '.disconnect-overlay' );
+		this.disconnectText = {
+			general: document.querySelector( '.disconnect-overlay .dialog-text span.general' ),
+			invalidAmp: document.querySelector( '.disconnect-overlay .dialog-text span.invalid-amp' ),
+		};
+		this.disconnectButtons = {
+			continue: document.querySelector( '.disconnect-overlay .button.continue' ),
+			exit: document.querySelector( '.disconnect-overlay .button.exit' ),
+			goBack: document.querySelector( '.disconnect-overlay .button.go-back' ),
+		};
+		this.addDisconnectButtonListeners();
+
+		// Load clients.
+		Promise.all( this.getIframeLoadedPromises() );
+	}
+
+	/**
+	 * Add event listeners for buttons on disconnect overlay.
+	 */
+	addDisconnectButtonListeners() {
+		// The 'Continue' button simply hides the 'disconnected' overlay.
+		this.disconnectButtons.continue.addEventListener( 'click', () => {
+			this.disconnectOverlay.classList.remove( 'disconnected' );
+		} );
+
+		// The 'Exit' button navigates the parent window to the non-AMP version of the current URL.
+		this.disconnectButtons.exit.addEventListener( 'click', () => {
+			window.location.assign( this.exitLink.href );
+		} );
+
+		/*
+		 * The 'Go back' button goes back to the previous page of the parent window.
+		 */
+		this.disconnectButtons.goBack.addEventListener( 'click', () => {
+			window.history.back();
+		} );
+	}
+
+	/**
+	 * Return promises to load iframes asynchronously.
+	 *
+	 * @return {[Promise<Function>, Promise<Function>]} Promises that determine if the iframes are loaded.
+	 */
+	getIframeLoadedPromises() {
+		return [
+			new Promise( ( resolve ) => {
+				this.nonAmpIframe.addEventListener( 'load', () => {
+					this.toggleDisconnectOverlay( this.nonAmpIframe );
+					resolve();
+				} );
+			} ),
+
+			new Promise( ( resolve ) => {
+				this.ampIframe.addEventListener( 'load', () => {
+					this.toggleDisconnectOverlay( this.ampIframe );
+					resolve();
+				} );
+			} ),
+		];
+	}
+
+	/**
+	 * Validates whether or not the window document is AMP compatible.
+	 *
+	 * @param {Document} doc Window document.
+	 * @return {boolean} True if AMP compatible, false if not.
+	 */
+	documentIsAmp( doc ) {
+		return doc.documentElement.hasAttribute( 'amp' ) ||
+			doc.documentElement.hasAttribute( '⚡️' );
```

Comment:
I think this check needs to also look for whether the `doc.querySelector( 'head > script[src="https://cdn.ampproject.org/v0.js]' )` is true. This will be more robust to handle the case of where we intentionally serve the AMP framework in a non-AMP page. Granted, this is only done in Standard mode at present, but it would be good to include here for completeness.